content/concepts/transport.md

```diff
@@ -115,3 +115,98 @@ and some places in the codebase still use the "swarm" terminology.
+   HoL blocking occurs when a the head packet holds up the other packets in the transmission 
+   queue. 
+   {{% /notice %}}
+
+2. Ossification: Because TCP is unencrypted, middleboxes can inspect and modify
+   TCP header fields and may break unexpectedly when they encounter anything they don’t like.
+
+   {{% notice "info" %}}
+   Middleboxes are intermediary networking devices that can perform special functions to 
+   inspect, transform, and even manipulate network traffic. Examples of middleboxes are 
+   firewalls, NATs, proxies, and load balancers.
+   {{% /notice %}}
+
+3. Handshake inefficiency: the 3-way handshake is inefficient, as it spends 1-RTT on verifying 
+   the client’s address.
+
+   {{% notice "info" %}}
+   TCP requires a 3-way handshake to establish a reliable, bidirectional connection as it intends to.
+   {{% /notice %}}
+
+We need a transport protocol that:
+
+- Understands streams 
+- Overcomes HOL blocking (Head-of-line blocking)
+- Overcomes the latency of connection setup
+- Overcomes the ossification risks of TCP
+
+and, ideally, keeps the guarantees of TCP.
+
+A web browser connection typically entails the following (TCP+TLS+HTTP/2):
+
+1. Transport layer: TCP runs on top of the IP layer to provide a reliable 
+   byte stream.
+   - TCP provides a reliable, bidirectional connection between two end systems.
+2. Secure communication layer: A TLS handshake runs on top of TCP to,
+   establishing an encrypted and authenticated connection.
+   - Standard TLS over TCP requires 3-RTT. A typical TLS 1.3 handshake takes 1-RTT.
+3. Application layer: HTTP runs on a secure transport connection to transfer 
+   information and applies a stream muxer to serve multiple requests.
+   - Application data starts to flow.
+
+<!-- to add diagram -->
+
+### What is QUIC?
+
+QUIC combines the functionality of these layers: it sends UDP packets. Therefore, 
+it is responsible for loss detection and repair itself. By using encryption, 
+QUIC avoid middleboxes. The TLS 1.3 handshake is performed in the first flight, 
+removing the 1-RTT cost of verifying the client’s address. QUIC also exposes multiple 
+streams, so no stream multiplexer is needed at the application layer. Part of the application 
+layer is also built directly into QUIC; when you run HTTP on top of QUIC; only a small shim 
+layer exists that maps 
+[HTTP semantics](https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html) 
+onto QUIC streams.
+
+QUIC supports the resumption of connections (0-RTT connections), allowing a 
+client to send application data right away, even before the QUIC handshake has finished.
+
+<!-- to add diagram -->
+
+### QUIC in libp2p
+
+libp2p only supports bidirectional streams and uses TLS 1.3 by default. 
+The streams in libp2p map cleanly to QUIC streams.
+
+When a connection starts, peers will take their host key and create a self-signed CA 
+certificate. They then sign an intermediate chain using their self-signed CA and put it 
```

Comment:
Unsure about this, but is it still a **CA** (= Certificate authority) certificate if it is self-signed? @marten-seemann?

Reply:
You can set up your own CA and use it to self-sign - my understanding is that as QUIC only works with secure connections, the certificate needs to be issued by a CA instead of it just being self-signed

Reply:
> the certificate needs to be issued by a CA instead of it just being self-signed

On the x509 level, that's the same thing. Any certificate that signs another certificates is a CA certificate. That said, browsers (and many other TLS implementaions) will (in their default setting) only accept CAs from a list of trusted CAs, either shipped with the OS or the browser.

QUIC doesn't introduce any special requirements here. The properties are exactly what they are for a normal TLS 1.3 handshake. QUIC just define a way to run a handshake that doesn't do a TLS 1.3 handshake.